DOI.js

```diff
@@ -20,7 +20,17 @@ var selectArray = {};
 
 
 // builds a list of DOIs
-function getDOIs(doc) {
+function getDOIs(doc, url) {
+	var dois = [], m;
+	
+	// Extract DOIs from the current URL
+	var rx = /10.[0-9]{4,}?\/[^\s&"'?#,]*[^\s&"'?#\/.,]/g;
```

Comment:
We usually use `re` for this, not `rx`.

Also, seems like we don't need to exclude quotes in this context.

And maybe we do need to URL decode? I'm not sure whether a URL that passes through here is necessarily decoded.